File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        l = 0
        r = len(s)-1
        while l < r:
            print(l,r)
            print(s[l],s[r])
            if s[l].isalnum() is False:
                l+=1
            elif s[r].isalnum() is False: 
                r-=1
            else:
                if s[l] == s[r]:
                    l+=1
                    r-=1
                else:
                    return False
        return True

File: Data Structures & Algorithms/palindrome-partitioning/submission-1.cpp
#include <vector>
class Solution {
    vector<vector<string>> res;
public:
    vector<vector<string>> partition(string s) {
        vector<string> curr;
        dfs(s,curr);
        return res;
    }

    void dfs(string s, vector<string> &curr){
        if (s.size()==0){
            res.push_back(curr);
        }

        for (int i=1;i<=s.size();i++){
            string curr_str = s.substr(0,i);
            if (is_Palindrome(curr_str)==true){
                curr.push_back(curr_str);
                dfs(s.substr(i,s.size()-i),curr);
                curr.pop_back();
            }
        }

    }


    bool is_Palindrome(const string &s){
        auto rev = s.end();
        rev--;
        for (auto it=s.begin();it<rev;it++){
            if (*it != *rev) return false;  
            rev--; 
        } 
        return true;
    }
};
